crates/pico_structures/src/world.rs

```diff
@@ -97,4 +298,41 @@ impl World {
             .get(index)
             .map(|v| v.as_slice())
     }
+
+    /// Get pre-calculated sky light data for a chunk column.
+    /// Returns a slice of LightSection, one for each Y section in the chunk.
+    pub fn get_chunk_sky_light(&self, chunk_x: i32, chunk_z: i32) -> Option<&[LightSection]> {
+        if chunk_x < 0
+            || chunk_x >= self.size_in_chunks.x()
+            || chunk_z < 0
+            || chunk_z >= self.size_in_chunks.z()
```

Comment:
This out of bound check, which is used in at least 3 methods, could be refactored to reduce the duplicated code.